### PR TITLE
Expand operations for Mpi

### PR DESCRIPTION
**Pull request by jack-fortanix**
Taken from internal version, `rustfmt`ed, and some adjustments for different feature flags.

### CODE REVIEW

File: mbedtls/src/lib.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ mod wrapper_macros;
 //      API
 // ==============
 #[allow(dead_code)] // to be exported once the API is more complete
```
Comment (jethrogb) [inline mbedtls/src/lib.rs]: I think this lint should go now

Comment (jack-fortanix) [reply mbedtls/src/lib.rs]: Done

**Comment by jack-fortanix** [comment]
A bit of code used  `trim_right_matches`, but this is deprecated starting in 1.33. Since warnings are errors, I changed it to use `trim_end_matches`. But that new API doesn't exist in the old nightly used in the no-`std` tests. What to do?

**Comment by jethrogb** [comment]
Use `trim_right_matches` and apply `#[allow(deprecated)]` in strategic places.

**Comment by jethrogb** [comment]
r=me after you get CI to pass

**Comment by jack-fortanix** [comment]
CI happy now

**Pull request merged**